shanoir-ng-import/src/main/java/org/shanoir/ng/importer/ImporterApiController.java

```diff
@@ -777,20 +777,25 @@ public int compare(File f1, File f2) {
 				if (job.getPatients().get(0).getStudies().get(0).getSeries().get(0).getEquipment().getDeviceSerialNumber() != null) {
 					equipmentIdFromDicom = (Long) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.EQUIPMENT_FROM_CODE_QUEUE, job.getPatients().get(0).getStudies().get(0).getSeries().get(0).getEquipment().getDeviceSerialNumber());
 					if (equipmentIdFromDicom != null) {
-						Properties props = new Properties();
-						props.setProperty("EQUIPMENT_ID_PROPERTY", "" + equipmentIdFromDicom);
-						props.setProperty("STUDY_ID_PROPERTY", "" + studyId);
-						props.setProperty("STUDYCARD_ID_PROPERTY", "" + studyCardId);
-
-						Long newStudyCardId = (Long) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.IMPORT_STUDY_CARD_QUEUE, props);
-						if (newStudyCardId != null)  {
-							studyCardId = newStudyCardId;
+						if (studyCardId != null) {
+							Properties props = new Properties();
+							props.setProperty("EQUIPMENT_ID_PROPERTY", "" + equipmentIdFromDicom);
+							props.setProperty("STUDY_ID_PROPERTY", "" + studyId);
+							props.setProperty("STUDYCARD_ID_PROPERTY", "" + studyCardId);
+
+							Long newStudyCardId = (Long) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.IMPORT_STUDY_CARD_QUEUE, props);
+							if (newStudyCardId != null) {
+								studyCardId = newStudyCardId;
+							}
 						}
 						job.setAcquisitionEquipmentId(equipmentIdFromDicom);
+					} else {
+						job.setAcquisitionEquipmentId(equipmentId);
 					}
 				}
 
-				job.setStudyCardId(studyCardId);
+				if (studyCardId != null)
```

Comment:
I wonder if this two if (studyCardId != null) { should better be if (studyCardId != 0L) {